### PR TITLE
[#85542806] show warnings when the attribute value type coercion throws ...

### PR DESCRIPTION
**Pull request by SpecialDOS**
...an exception (specifically to improve the error reporting when json values are not valid json)

### CODE REVIEW

**Comment by mcarlson** [comment]
Don't exceptions cause poor performance? Maybe this should only be happening in debug mode. It seems like it's in a fairly performance critical area!

**Comment by SpecialDOS** [comment]
Good call, I'll turn it on only for debug

File: layout.coffee [hunk 1]
```diff
@@ -284,10 +284,16 @@ window.dr = do ->
     _coerceType: (name, value, type) ->
       type ||= @types[name]
       if type
-        unless (typemappings[type])
-          showWarnings ["Invalid type '#{type}' for attribute '#{name}', must be one of: #{Object.keys(typemappings).join(', ')}"]
-          return
-        value = typemappings[type](value)
+        if debug || test
```
Comment (mcarlson) [inline layout.coffee]: Consider 'debug or test' to be more CS-ey

**Comment by mcarlson** [comment]
:+1:

**Pull request merged**